code/modules/surgery/operations/_operation.dm

```diff
@@ -1055,7 +1114,7 @@ GLOBAL_DATUM_INIT(operations, /datum/operation_holder, new)
  */
 /datum/surgery_operation/proc/on_preop(atom/movable/operating_on, mob/living/surgeon, tool, list/operation_args)
 	PROTECTED_PROC(TRUE)
-	var/mob/living/patient = get_patient(operating_on)
+	var/mob/living/patient = get_patient(operating_on) || operating_on
```

Comment:
I think this is definitely erroneous, you're allowing non-mobs to be passed into `display_results` `target` which will runtime 

Also you don't do this in `on_success` and `on_failure`